Carmen Maranhas Gomez-Choco

## Descargamos el archivo en csv
Al abrirlo en excel se observa una estructura organizada en una tabla con sus columnas identificadas correctamente. 
Tambien observamos como algunas celdas aparecen incompletas (no poseen todos los campos rellenos)
importamos la libreria de pandas para poder realizar la practica

In [29]:
import pandas as pd

https://datos.gob.es/es/catalogo/a08002880-fiestas-de-interes-turistico-declaradas-en-castilla-la-mancha
Este he cambiado el nombre la las columna sporque eran muy largas, las he cmabiado en la hoja excell

In [19]:
Turismo_data = pd.read_csv("Datos/FIESTAS.csv", sep = ";" )
Turismo_data

DENOMINACIÓN             LOCALIDAD    PROVINCIA  \
0             SEMANA SANTA Y TAMBORADA                AGRAMÓN     ALBACETE   
1             FIESTA DE JUDAS Y MUÑECAS              ALAMILLO  CIUDAD REAL   
2    CABALGATA AÉREA DE LOS REYES MAGOS              ALARILLA  GUADALAJARA   
3                     FERIA DE ALBACETE              ALBACETE     ALBACETE   
4                          SEMANA SANTA              ALBACETE     ALBACETE   
..                                  ...                   ...          ...   
144                 FIESTA DE LAS PACES  VILLARTA DE SAN JUAN  CIUDAD REAL   
145              JORNADAS CALDERONIANAS                 YEPES       TOLEDO   
146      FERIA DE TRADICIONES POPULARES                 YESTE     ALBACETE   
147            ROMERÍA DE SAN BARTOLOMÉ                 YESTE     ALBACETE   
148                     TORO ENMAROMADO                YUNCOS       TOLEDO   

                FECHA DE CELEBRACION DE LA FIESTA  FECHA_Resolucion  \
0         Fecha variable según calendario nacional       23/02/2016   
1                                 Sábado de gloria       21/03/2022   
2                                       5 de enero       20/12/2018   
3                               7-17 de septiembre       31/07/2008   
4         Fecha variable según calendario nacional       28/03/2017   
..                                             ...              ...   
144                                 23-26 de enero       28/03/1994   
145  Fecha variable. En función del Corpus Christi       20/05/2015   
146                               20-22 de octubre       12/12/2014   
147                                   24 de agosto       18/07/2022   
148                                        Febrero       10/07/2017   

    FIESTAS_de_interes_turistico FIESTAS_INTERES_REGIONAL   FECHA_BOE  \
0                            NaN               26/02/2016         NaN   
1                            NaN               29/03/2022         NaN   
2                            NaN               28/12/2018         NaN   
3                            NaN                        X  16/02/1980   
4                            NaN               24/10/1997  31/03/2017   
..                           ...                      ...         ...   
144                          NaN               15/04/1994         NaN   
145                          NaN               01/06/2015         NaN   
146                          NaN               26/12/2014         NaN   
147                          NaN               26/07/2022         NaN   
148                          NaN               27/07/2017         NaN   

    FECHA_ACTUALIZACION_BOE                                      OBSERVACIONES  
0                       NaN  De Domingo de Ramos a Domingo de Resurrección,...  
1                       NaN                                                NaN  
2                       NaN                                                NaN  
3                01/09/2008                                                NaN  
4                       NaN  De Domingo de Ramos a Domingo de Resurrección,...  
..                      ...                                                ...  
144                     NaN                                                NaN  
145                     NaN                     Segundo fin de semana de junio  
146                     NaN                                                NaN  
147                     NaN                                                NaN  
148                     NaN                                                NaN  

[149 rows x 10 columns]

In [23]:
Turismo_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 149 entries, 0 to 148
Data columns (total 10 columns):
 #   Column                              Non-Null Count  Dtype 
---  ------                              --------------  ----- 
 0   DENOMINACIÓN                        149 non-null    object
 1   LOCALIDAD                           149 non-null    object
 2   PROVINCIA                           149 non-null    object
 3   FECHA DE CELEBRACION DE LA FIESTA   149 non-null    object
 4   FECHA_Resolucion                    149 non-null    object
 5   FIESTAS_de_interes_turistico        12 non-null     object
 6   FIESTAS_INTERES_REGIONAL            143 non-null    object
 7   FECHA_BOE                           21 non-null     object
 8   FECHA_ACTUALIZACION_BOE             6 non-null      object
 9   OBSERVACIONES                       77 non-null     object
dtypes: object(10)
memory usage: 11.8+ KB


La info del DataFrame Turismo de Castilla la Mancha  nos indican que tiene 10 campos  con 149 registros o entradas 
Los datos fiesta de inters turistico (12) y fecha de actualizacion boe(6) tienen muy poca informacion 

#https://datos.gob.es/es/catalogo/l01280796-alumbrado-navideno
#El fichjeoo que descarga se llamade otra forma pero lo he renombrado porque creia que era ese el problema

In [24]:
alumbrado_data = pd.read_csv("Datos/ALUMBRADO.csv", sep = ";" )
alumbrado_data

POSICION           MOTIVO  \
0                                             GRAN VIA  ARCOS COMPACTOS   
1                                         PLAZA CHUECA  MOTIVO SINGULAR   
2                                              MONTERA  ARCOS COMPACTOS   
3             FUENCARRAL - DE GRAN VIA A HERNAN CORTES  ARCOS COMPACTOS   
4               HORTALEZA. DE GRAN VIA A HERNAN CORTES  ARCOS COMPACTOS   
..                                                 ...              ...   
240                   ENTRADA PLAZA MAYOR POR C TOLEDO  MOTIVO NAVIDEÑO   
241             PASEO DEL PRADO (DE CIBELES A NEPTUNO)        CADENETAS   
242  PASEO DEL PRADO, ENTRE PLAZA DE NEPTUNO Y GLOR...        CADENETAS   
243  PASEO DE RECOLETOS - CALZADA (DE CIBELES A COLON)        CADENETAS   
244             PARQUE DEL RETIRO - PERIMETRO ESTANQUE          CEREZOS   

                                            COMENTARIO  idelem NOVEDAD  \
0               ARCOS CON TRES VANOS, INCLUYENDO ACERA       1      NO   
1                    CATENARIAS CON COLGANTES CRUZADOS       2      SI   
2    MOTIVOS DE RENOS, CAMPANAS, PATINES, BOLAS, MU...       3      NO   
3                                    ARCOS CON LEMURES       4      NO   
4                        ARCOS CON CIRCULOS DE COLORES       5      NO   
..                                                 ...     ...     ...   
240                                         NACIMIENTO     252      NO   
241                                CORDON DE BOMBILLAS     253      NO   
242  CORDON DE BOMBILLAS, EN EL LADO DEL ARBOLADO D...     254      NO   
243                                      HILO LUMINOSO     255      NO   
244                         16 CEREZOS, VARIOS COLORES     257      NO   

            DISEÑO       DISTRITOS                 BARRIO     TIPO  LONGITUD  \
0       BEN BUSCHE          Centro                 Cortes   LINEAL    1336.0   
1      ANA LOCKING          Centro               Justicia   LINEAL     146.0   
2      MAYA HANSEN          Centro                    Sol   LINEAL     291.0   
3    TERESA HELBIG          Centro               Justicia   LINEAL     393.0   
4                           Centro               Justicia   LINEAL     388.0   
..             ...             ...                    ...      ...       ...   
240     BEN BUSCHE          Centro                    Sol  PUNTUAL       NaN   
241                 Centro, Retiro  Cortes, Los Jerónimos   LINEAL     373.0   
242                 Centro, Retiro  Cortes, Los Jerónimos   LINEAL     606.0   
243                 Centro, Retiro  Cortes, Los Jerónimos   LINEAL     562.0   
244                         Retiro          Los Jerónimos  PUNTUAL       NaN   

     N_ELEMENTOS                  PROMOTOR  \
0            NaN  AG OBRAS Y EQUIPAMIENTOS   
1            NaN  AG OBRAS Y EQUIPAMIENTOS   
2            NaN  AG OBRAS Y EQUIPAMIENTOS   
3            NaN  AG OBRAS Y EQUIPAMIENTOS   
4            NaN  AG OBRAS Y EQUIPAMIENTOS   
..           ...                       ...   
240          NaN  AG OBRAS Y EQUIPAMIENTOS   
241          NaN  AG OBRAS Y EQUIPAMIENTOS   
242          NaN  AG OBRAS Y EQUIPAMIENTOS   
243          NaN  AG OBRAS Y EQUIPAMIENTOS   
244          4.0  AG OBRAS Y EQUIPAMIENTOS   

                                                 TEXTO  
0    La instalación lumínica se inspira en las imág...  
1                                      MOTIVO LUMINOSO  
2    “He querido mezclar elementos tradicionales co...  
3    “Con los lémures siempre hemos sentido una con...  
4                                      MOTIVO LUMINOSO  
..                                                 ...  
240                                    MOTIVO LUMINOSO  
241  CADENETAS DE 22 M CON 40 PORTALAMPARAS  DE TIP...  
242  CADENETAS DE 22 M CON 40 PORTALAMPARAS  DE TIP...  
243                                    MOTIVO LUMINOSO  
244                                    MOTIVO LUMINOSO  

[245 rows x 13 columns]

In [26]:
alumbrado_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 245 entries, 0 to 244
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   POSICION     245 non-null    object 
 1   MOTIVO       245 non-null    object 
 2   COMENTARIO   245 non-null    object 
 3   idelem       245 non-null    int64  
 4   NOVEDAD      245 non-null    object 
 5   DISEÑO       244 non-null    object 
 6   DISTRITOS    244 non-null    object 
 7   BARRIO       244 non-null    object 
 8   TIPO         245 non-null    object 
 9   LONGITUD     159 non-null    float64
 10  N_ELEMENTOS  5 non-null      float64
 11  PROMOTOR     245 non-null    object 
 12  TEXTO        245 non-null    object 
dtypes: float64(2), int64(1), object(10)
memory usage: 25.0+ KB


Alumnbrado Navideño tiene 13 campos y 245 registros. Escepto el campo n_elementos (5) y longitud (159) todos tienen la informacion completa

https://datos.gob.es/es/catalogo/ea0010587-gasto-medio-por-centro-segun-comunidad-autonoma-identificador-api-t25-p454-e01-a2022-2ccaa_9890-1car_9891-l0-0201011-px
Este fichero no he modificado nada y es en el que me he dado cuenta que había que poner el separador

In [15]:
gasto_data = pd.read_csv("Datos/59739.csv", sep = ";" )
gasto_data

Total Nacional Comunidades y Ciudades Autónomas  \
0   Total Nacional                              NaN   
1   Total Nacional                     01 Andalucía   
2   Total Nacional                        02 Aragón   
3   Total Nacional       03 Asturias, Principado de   
4   Total Nacional                04 Balears, Illes   
5   Total Nacional                      05 Canarias   
6   Total Nacional                     06 Cantabria   
7   Total Nacional               07 Castilla y León   
8   Total Nacional          08 Castilla - La Mancha   
9   Total Nacional                      09 Cataluña   
10  Total Nacional          10 Comunitat Valenciana   
11  Total Nacional                   11 Extremadura   
12  Total Nacional                       12 Galicia   
13  Total Nacional          13 Madrid, Comunidad de   
14  Total Nacional             14 Murcia, Región de   
15  Total Nacional   15 Navarra, Comunidad Foral de   
16  Total Nacional                    16 País Vasco   
17  Total Nacional                     17 Rioja, La   
18  Total Nacional                         18 Ceuta   
19  Total Nacional                       19 Melilla   

                             Gasto      Total  
0   Gasto medio por centro (euros)    348.804  
1   Gasto medio por centro (euros)    275.190  
2   Gasto medio por centro (euros)    368.301  
3   Gasto medio por centro (euros)    395.891  
4   Gasto medio por centro (euros)    318.731  
5   Gasto medio por centro (euros)    445.153  
6   Gasto medio por centro (euros)    573.320  
7   Gasto medio por centro (euros)    235.180  
8   Gasto medio por centro (euros)    338.167  
9   Gasto medio por centro (euros)    479.998  
10  Gasto medio por centro (euros)    334.160  
11  Gasto medio por centro (euros)    344.144  
12  Gasto medio por centro (euros)    236.134  
13  Gasto medio por centro (euros)    777.764  
14  Gasto medio por centro (euros)    222.489  
15  Gasto medio por centro (euros)    642.366  
16  Gasto medio por centro (euros)    275.623  
17  Gasto medio por centro (euros)    356.521  
18  Gasto medio por centro (euros)  3.007.663  
19  Gasto medio por centro (euros)  1.697.448

In [28]:
gasto_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 4 columns):
 #   Column                            Non-Null Count  Dtype 
---  ------                            --------------  ----- 
 0   Total Nacional                    20 non-null     object
 1   Comunidades y Ciudades Autónomas  19 non-null     object
 2   Gasto                             20 non-null     object
 3   Total                             20 non-null     object
dtypes: object(4)
memory usage: 768.0+ bytes


El dataFrame gasto por comunidad autonoma tiene cuatro campos y estan todas las filas rellenas excepto comunidades autonomas que falta una

In [32]:
#rango de valores
gasto_data.index

RangeIndex(start=0, stop=20, step=1)

In [34]:
#nombre de las columnas
gasto_data.columns

Index(['Total Nacional', 'Comunidades y Ciudades Autónomas', 'Gasto', 'Total'], dtype='object')

In [43]:
gasto_data[["Total"]].max()

Total    777.764
dtype: object

In [44]:
gasto_data[["Total"]].min()

Total    1.697.448
dtype: object